### PR TITLE
empty node name logged by node_status_updater

### LINKED ISSUE #37777: Investigate empty node name logged by node_status_updater
**Kubernetes version** (use `kubectl version`): 1.5

**Environment**:
- **Cloud provider or hardware configuration**: GCE
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:
Node status updater is trying to look for a node name with empty string. Investigation is needed as to why this happens.
https://gist.github.com/rkouj/979f70c1d3fda1b1ca5d751b65009ff9#file-gistfile1-txt-L12

**How to reproduce it** (as minimally and precisely as possible):
I found it when I was running the manual upgrade test for 1.4 to 1.5
However, I suspect this is something that happens frequently.

### PR DESCRIPTION
**Pull request by yastij**
**What this PR does / why we need it**:

**Which issue this PR fixes** : fixes #37777

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by justinsb** [comment]
Do we know why this is happening?

**Comment by yastij** [comment]
@justinsb - still auditing code to find out the root cause, this is a temporary fix.

**Comment by yastij** [comment]
@justinsb just had the time to work on this one today, got stuck on other k8s issues. still trying to investigate the cause of the error, but getting close : 

```go
func (nsu *nodeStatusUpdater) UpdateNodeStatuses() 
```
calls to `nsu.actualStateOfWorld#GetVolumesToReportAttached()` returns a map from actual state of world, which contains nodes that needs status update, this map somehow contains an entry with empty nodeName.  

The problem now is to know how our representation of node could diverge that way (actual nodeName vs nodes to update in the underlying nodeInformer store). @rkouj when attaching volumes to node, noticed anything strange, like node missing mounted volumes ? if not the workaround provided is sufficient as temporary solution. ping @saad-ali 

I'm also trying to see if it is related to some upgrade unknown effects (trying to see if this is still happening on upgrades from 1.5 to 1.6 and from 1.6 to 1.7).

File: pkg/controller/volume/attachdetach/attach_detach_controller.go [hunk 1]
```diff
@@ -427,12 +427,14 @@ func (adc *attachDetachController) podDelete(obj interface{}) {
 
 func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	node, ok := obj.(*v1.Node)
-	// TODO: investigate if nodeName is empty then if we can return
-	// kubernetes/kubernetes/issues/37777
+
```
Comment (xiangpengzhao) [inline pkg/controller/volume/attachdetach/attach_detach_controller.go]: nit: this line can be deleted.

File: pkg/controller/volume/attachdetach/attach_detach_controller.go [hunk 2]
```diff
@@ -443,12 +445,15 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 
 func (adc *attachDetachController) nodeUpdate(oldObj, newObj interface{}) {
 	node, ok := newObj.(*v1.Node)
-	// TODO: investigate if nodeName is empty then if we can return
+
```
Comment (xiangpengzhao) [inline pkg/controller/volume/attachdetach/attach_detach_controller.go]: this line too.

**Pull request closed without merging**